main/src/main/scala/sbt/Main.scala

```diff
@@ -940,13 +940,17 @@ object BuiltinCommands {
 
     val session = Load.initialSession(structure, eval, s0)
     SessionSettings.checkSession(session, s2)
-    val s3 = addCacheStoreFactoryFactory(Project.setProject(session, structure, s2))
+    val s3 = Project.setProject(
+      session,
+      structure,
+      s2,
+      st => setupGlobalFileTreeRepository(addCacheStoreFactoryFactory(st))
+    )
     val s4 = s3.put(Keys.useLog4J.key, Project.extract(s3).get(Keys.useLog4J))
-    val s5 = setupGlobalFileTreeRepository(s4)
```

Comment:
Did you try just making this s3 and passing s3 into Project.setProject instead of s2? Based on the description, it seems like that would be a simpler fix but I might be missing something.

Reply:
The whole numbered `sn` thing is not great. It would be better if this had had a descriptive name like `stateWithFileRepository` rather than `s5`. Refactoring would be easier and it would be easier to understand.

Reply:
> Did you try just making this s3 and passing s3 into Project.setProject instead of s2? Based on the description, it seems like that would be a simpler fix but I might be missing something.

That would would fix _my_ problem, however it would not be correct, because than Project.setProject would call `onUnload` of the "old lifecycle" with the new objects which are meant to be used starting with `onLoad` in the "new lifecycle". So no, this is not a solution.